cmd/importaccount.go

```diff
@@ -0,0 +1,184 @@
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/nats-io/jwt"
+	"github.com/nats-io/nkeys"
+	"github.com/nats-io/nsc/cmd/store"
+	"github.com/spf13/cobra"
+)
+
+func createImportAccountCmd() *cobra.Command {
+	var params ImportAccount
+	cmd := &cobra.Command{
+		Use:          "account --jwt <account-jwt>",
+		Short:        "Imports an account from a jwt file and resign with operator if self signed",
+		Example:      `nsc import account --jwt <account-jwt>`,
+		Args:         MaxArgs(0),
+		SilenceUsage: false,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := RunMaybeStorelessAction(cmd, args, &params); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	cmd.Flags().BoolVarP(&params.skip, "skip", "", false, "skip validation issues, they can be edited out")
+	cmd.Flags().BoolVarP(&params.overwrite, "overwrite", "", false, "overwrite existing account")
+	cmd.Flags().StringVarP(&params.file, "jwt", "j", "", "account jwt to import")
+	cmd.MarkFlagRequired("jwt")
+	return cmd
+}
+
+func init() {
+	importCmd.AddCommand(createImportAccountCmd())
+}
+
+type fileImport struct {
+	file      string
+	skip      bool
+	overwrite bool
+}
+
+type ImportAccount struct {
+	SignerParams
+	fileImport
+}
+
+func (p *ImportAccount) SetDefaults(ctx ActionCtx) error {
+	p.SignerParams.SetDefaults(nkeys.PrefixByteOperator, true, ctx)
+	return nil
+}
+
+func (p *ImportAccount) PreInteractive(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *ImportAccount) Load(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *ImportAccount) PostInteractive(ctx ActionCtx) error {
+	signers, err := GetOperatorSigners(ctx)
+	if err != nil {
+		return err
+	}
+	p.SignerParams.SetPrompt("select the key to sign the account")
+	return p.SignerParams.SelectFromSigners(ctx, signers)
+}
+
+func (p *ImportAccount) Validate(ctx ActionCtx) error {
+	fi, err := os.Stat(p.file)
+	if err != nil {
+		return err
+	}
+	if fi.IsDir() {
+		return fmt.Errorf("%#q is a directory", p.file)
+	}
+	if err := p.SignerParams.Resolve(ctx); err != nil {
+		return err
+	}
+	if !ctx.StoreCtx().Store.IsManaged() {
+		signers, err := GetOperatorSigners(ctx)
+		if err != nil {
+			return err
+		}
+		ok, err := ValidSigner(p.SignerParams.signerKP, signers)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return errors.New("invalid account signer")
+		}
+	}
+
+	return nil
+}
+
+func (p *ImportAccount) Run(ctx ActionCtx) (store.Status, error) {
+	r := store.NewDetailedReport(true)
+	theJWT, err := ioutil.ReadFile(p.file)
+	if err != nil {
+		r.AddError("failed to import %#q: %v", p.file, err)
+		return r, err
+	}
+	claim, err := jwt.DecodeAccountClaims(string(theJWT))
+	if err != nil {
+		r.AddError("failed to decode %#q: %v", p.file, err)
+		return r, err
+	}
+	if validateAndReport(claim, p.skip, r) {
+		return r, nil
+	}
+	if ctx.StoreCtx().Store.HasAccount(claim.Name) {
+		if !p.overwrite {
+			r.AddError("Account already exists, overwrite with --overwrite")
```

Comment:
all status messages should start with a lower case to match other commands unless it is the name of some entity.